File: sample/src/main/java/com/github/lany192/box/sample/ui/splash/SplashViewModel.java
package com.github.lany192.box.sample.ui.splash;

import androidx.lifecycle.MutableLiveData;

import com.github.lany192.box.mvvm.LifecycleViewModel;

import javax.inject.Inject;

import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class SplashViewModel extends LifecycleViewModel {
    private final MutableLiveData<String> welcome = new MutableLiveData<>();

    @Inject
    public SplashViewModel() {
        welcome.postValue("Hello Box");
    }

    public MutableLiveData<String> getWelcome() {
        return welcome;
    }
}

File: sample/src/main/java/com/github/lany192/box/sample/ui/splash/SplashViewModel.kt
package com.github.lany192.box.sample.ui.splash

import androidx.lifecycle.MutableLiveData
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject
import com.github.lany192.arch.viewmodel.LifecycleViewModel

@HiltViewModel
class SplashViewModel @Inject constructor() : LifecycleViewModel() {
    val welcome = MutableLiveData<String>()

    init {
        welcome.postValue("Hello Box")
    }
}